pkg/commands/mcp/actions/mcp.go

```diff
@@ -0,0 +1,302 @@
+			}
+
+			utils.McpLogf("Ready to receive requests...")
+
+			return runSTDIOServer(endpoint, httpClient)
+		},
+	}
+
+	return cmd
+}
+
+func setupMCPServer(cmd *cobra.Command) (string, *http.Client, error) {
+	utils.McpLogf("=== Starting Setup ===")
+
+	sessionCfg, err := config.GetProfileByName("")
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get profile: %w", err)
+	}
+
+	apiURL, err := url.Parse(sessionCfg.ApiURL)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to parse API URL: %w", err)
+	}
+	apiURL.Path = McpEndpointPath
+
+	var httpClient *http.Client
+	if sessionCfg.PersonalToken != "" {
+		httpClient = aponoapi.HTTPClientWithPersonalToken(sessionCfg.PersonalToken)
+	} else {
+		client, err := aponoapi.CreateClient(cmd.Context(), "")
+		if err != nil {
+			return "", nil, fmt.Errorf("failed to create API client: %w", err)
+		}
+		httpClient = client.APIClient.GetConfig().HTTPClient
+	}
+
+	utils.McpLogf("=== Setup Finished ===")
+	return apiURL.String(), httpClient, nil
+}
+
+func runSTDIOServer(endpoint string, httpClient *http.Client) error {
+	scanner := bufio.NewScanner(os.Stdin)
+
+	utils.McpLogf("=== STDIO Server Started, waiting for input ===")
+
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" {
+			continue
+		}
+
+		utils.McpLogf("Received Line: %s", line)
+
+		var request McpRequest
+		if err := json.Unmarshal([]byte(line), &request); err != nil {
+			utils.McpLogf("ERROR: Failed to parse JSON: %v", err)
+			response := McpResponse{
+				JSONRPC: JsonrpcVersion,
+				ID:      nil,
+				Error: map[string]interface{}{
+					"code":    ParseError,
+					"message": "Request received cannot be parsed as an MCP request",
+					"data":    fmt.Sprintf("Invalid JSON: %v", err),
+				},
+			}
+			sendResponse(response, EmptyErrorStatusCode)
+			continue
+		}
+
+		utils.McpLogf("Parsed request - Method: %s, ID: %v", request.Method, request.ID)
+
+		response, statusCode := sendMcpRequest(endpoint, httpClient, request)
+
+		if !isNotificationsMethod(request.Method) {
+			utils.McpLogf("Sending response - Status: %d, ID: %v", statusCode, response.ID)
+			sendResponse(response, statusCode)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		utils.McpLogf("ERROR: Scanner error: %v", err)
+		return fmt.Errorf("error reading stdin: %w", err)
+	}
+
+	return nil
+}
+
+func sendMcpRequest(endpoint string, httpClient *http.Client, request McpRequest) (McpResponse, int) {
+	utils.McpLogf("Sending request to endpoint: %s", endpoint)
+
+	requestBody, err := json.Marshal(request)
+	if err != nil {
+		utils.McpLogf("ERROR: Failed to marshal request: %v", err)
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    InternalError,
+				"message": "An internal error occurred when trying to send the MCP request to Apono",
+				"data":    "Failed to marshal request",
+			},
+		}, EmptyErrorStatusCode
+	}
```

Comment:
I dont see why you marshal the json again, you are creating a lot of garbage for no reason, just pass the original request the agent sent, along with the extracted id and method if needed